File: python/src/pre-recorded/transcription.py
import requests
import os
from time import sleep


def make_request(url, headers, method="GET", data=None, files=None):
    if method == "POST":
        response = requests.post(url, headers=headers, json=data, files=files)
    else:
        response = requests.get(url, headers=headers)
    return response.json()


print(os.getcwd())
file_path = "../../../data/anna-and-sasha-16000.wav"  # Change with your file path

if os.path.exists(file_path):  # This is here to check if the file exists
    print("- File exists")
else:
    print("- File does not exist")
    exit(0)


file_name, file_extension = os.path.splitext(
    file_path
)  # Get your audio file name + extension

with open(file_path, "rb") as f:  # Open the file
    file_content = f.read()  # Read the content of the file

headers = {
    "x-gladia-key": os.getenv("GLADIA_API_KEY", ""),  # Replace with your Gladia Token
    "accept": "application/json",
}

files = [("audio", (file_path, file_content, "audio/" + file_extension[1:]))]

print("- Uploading file to Gladia...")
upload_response = make_request(
    "https://api.gladia.io/v2/upload/", headers, "POST", files=files
)
print("Upload response with File ID:", upload_response)
audio_url = upload_response.get("audio_url")

data = {
    "audio_url": audio_url,
    "diarization": True,
}
# You can also send an URL directly without uploading it. Make sure it's the direct link and publicly accessible.
# For any parameters, please see: https://docs.gladia.io/reference/pre-recorded

headers["Content-Type"] = "application/json"

print("- Sending request to Gladia API...")
post_response = make_request(
    "https://api.gladia.io/v2/transcription/", headers, "POST", data=data
)

print("Post response with Transcription ID:", post_response)
result_url = post_response.get("result_url")

if result_url:
    while True:
        print("Polling for results...")
        poll_response = make_request(result_url, headers)

        if poll_response.get("status") == "done":
            print("- Transcription done: \n")
            print(poll_response.get("result"))
            break
        elif poll_response.get("status") == "error":
            print("- Transcription failed")
            print(poll_response)
        else:
            print("Transcription status:", poll_response.get("status"))
        sleep(1)

print("- End of work")

File: php/src/pre-recorded/transcription.php
<?php

$file_path = 'audio.mp3'; // Change with your file path

if (file_exists($file_path)) { // This is here to check if the file exists
    echo "- File exists\n";
} else {
    echo "- File does not exist\n";
}

$file_name = pathinfo($file_path, PATHINFO_FILENAME); // Get your audio file name
$file_extension = pathinfo($file_path, PATHINFO_EXTENSION); // Get your audio file extension

$audio_file = new CurlFile($file_path, 'audio/'.$file_extension); // Create a CurlFile object

$data = [
    // Sending a local audio file
    'audio' => $audio_file,
    // You can also send an URL for your audio file. Make sure it's the direct link and publicly accessible.
    // 'audio_url' => 'http://files.gladia.io/example/audio-transcription/split_infinity.wav',
    // Then you can pass any parameters you want. Please see: https://docs.gladia.io/api-reference/pre-recorded-flow
    'toggle_diarization' => true,
];

$headers = [
    'x-gladia-key: {YOUR_GLADIA_TOKEN}', // Replace with your Gladia Token
    'accept: application/json', // Accept json as a response, but we are sending a Multipart FormData
];

$curl = curl_init();
curl_setopt($curl, CURLOPT_URL, 'https://api.gladia.io/audio/text/audio-transcription/');
curl_setopt($curl, CURLOPT_POST, true);
curl_setopt($curl, CURLOPT_POSTFIELDS, $data);
curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);
curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);

echo "- Sending request to Gladia API...\n";
$response = curl_exec($curl);
$status_code = curl_getinfo($curl, CURLINFO_HTTP_CODE);

if ($status_code == 200) {
    echo "- Request successful\n";
    $result = json_decode($response, true);
    print_r($result);
} else {
    echo "- Request failed\n";
    $error = json_decode($response, true);
    print_r($error);
}

curl_close($curl);
echo "- End of work\n";

?>


